client/layout/masterbar/masterbar-cart/masterbar-cart-button.tsx

```diff
@@ -43,14 +58,26 @@ export function MasterbarCartButton( {
 		}
 	}, [ shouldShowCart, reduxDispatch ] );
 
+	useEffect( () => {
+		const isPricingPage = document.body.classList.contains( 'is-section-jetpack-cloud-pricing' );
+
+		//add the className for body, only if the page is in mobile view and in pricing page
+		if ( isActive && isMobile && isPricingPage ) {
+			document.body.classList.add( 'is-mobile-cart' );
```

Comment:
This feels a little awkward to me for three reasons: it's effectively using the body tag's class as a way to hold state, the class `is-mobile-cart` is very broadly named, and that the page this is designed to affect is in a very different part of the codebase from the masterbar.

I guess that you need to know on the pricing page if the masterbar cart is active for some reason. Is there a better way to do that?

What if we changed this `useEffect` to add a class which is specific to the masterbar cart, like `body--masterbar-cart-visible` (following BEM style), any time that the cart is visible; then the pricing page could respond to that class as needed without any page-specific logic here? If it also needs to know when the viewport is at mobile width it could use a media query (if inside CSS) or `useMobileBreakpoint` (if inside JS).

So something like:
```js
useEffect(() => {
	if (isActive) {
		document.body.classList.add( 'body--masterbar-cart-visible' );
	} else {
		document.body.classList.remove( 'body--masterbar-cart-visible' );
	}
}, [ isActive ]);

```

Reply:
@sirbrillig, thanks for this valuable comment. I have updated accordingly